### PR TITLE
feat(Google Cloud Firestore Node): Add support for service account and document creation with id

### PR DESCRIPTION
**Pull request by adrian-martinez-onestic**
## Summary

This PR adds two features:
- Allows to use Google Service Accounts for Firestore nodes authentication.
- Allows to create Firestore documents with specific ids (optionally).

## Details

While its possible to use Firestore using a service account, it is currently not supported by n8n, as noted at https://docs.n8n.io/integrations/builtin/credentials/google/#compatible-nodes. Allowing to use a service account in this node allows not depending on a personal account, and provides a reliable way to connect to Firestore.

In this way, proposed changes are backwards compatible, and offer the possibility of using either Oauth2 or Service Account authentication, and optionally, supply a document id, as it is possible while calling API or creating the document manually through the UI:

Captura de pantalla 2024-06-12 a las 12 43 00

Captura de pantalla 2024-06-12 a las 12 43 34

Captura de pantalla 2024-06-12 a las 12 47 35

Captura de pantalla 2024-06-12 a las 12 53 57

Captura de pantalla 2024-06-12 a las 12 56 35

Captura de pantalla 2024-06-12 a las 13 03 32

Captura de pantalla 2024-06-12 a las 13 03 49

### CODE REVIEW

**Review by Joffcom — DISMISSED** [review]
Tested and looks good, Thanks for this.

**Comment by netroy** [comment]
Since maintainers aren't allowed to edit this PR, I'm unable to fix the CI issues.
Can you please cherry-pick [this](https://github.com/n8n-io/n8n/commit/b76b8cab041e091f00f3ce5738c1a15978591dcb) commit, and merge the latest `master` back in 🙏🏽.
Once the CI is green, we can merge.

When creating any future PRs, please make sure that this checkbox is checked
Screenshot from 2024-06-26 13-32-22

**Comment by adrian-martinez-onestic** [comment]
Cherry-picked commit with CI fixes, also updated branch with latest master changes

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.50.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.50.0)

**Pull request merged**
